templates/pages/admin/customobjects/main.html.twig

```diff
@@ -137,6 +127,36 @@
             });
         </script>
     {% endif %}
+    <script type="module">
+        $('#mainformtable input[name="system_name"]').on('input', () => {
+            if ($('input[name="system_name"]').val() === '') {
+                $('input[name="system_name"]').data('manually_changed', false);
+            } else {
+                $('input[name="system_name"]').data('manually_changed', true);
+            }
+        });
+        function autoUpdateNameField() {
+            if ({{ not item.isNewItem() ? 'true' : 'false' }} || $('#mainformtable input[name="system_name"]').data('manually_changed')) {
+                return;
+            }
+            const reserved_names = {{ reserved_system_names|json_encode()|raw }}.map((n) => n.toLowerCase());
+            const existing_names = {{ existing_system_names|json_encode()|raw }}.map((n) => n.toLowerCase());
+            // label is made lowercase and spaces are replaced with underscores. All other characters are removed
+            $('#mainformtable input[name="system_name"]')
+                .val($('#mainformtable input[name="label"]').val().normalize('NFD').toLowerCase().replace(/ /g, '_').replace(/[^a-z_]/g, ''));
```

Comment:
I think we should let the ability to use uppercase chars here. For instance, our assets class names contains uppercase chars and we will probably want to transform them into generic assets in the future without having to change their system name case.

```suggestion
                .val($('#mainformtable input[name="label"]').val().normalize('NFD').replace(/ /g, '_').replace(/[^a-z_]/g, ''));
```